src/main/java/ru/ralsei/whatcanyousee/logic/MazeGameMap.java

```diff
@@ -0,0 +1,809 @@
+            currentX = cell.getX();
+            currentY = cell.getY();
+            cell.numberOfMonsters++;
+        }
+
+        /**
+         * Updates monster's state on gameplay tick -- for instance, monster could be moving each
+         * 5 ticks, so here he updates his counter.
+         */
+        abstract protected void updateOnTick();
+
+        int getImageId() {
+            return this.imageId;
+        }
+
+        protected void setImageId(int imageId) {
+            this.imageId = imageId;
+        }
+
+        protected int getInitialX() {
+            return initialX;
+        }
+
+        protected void setInitialX(int initialX) {
+            this.initialX = initialX;
+            currentX = initialX;
+        }
+
+        protected int getInitialY() {
+            return initialY;
+        }
+
+        protected void setInitialY(int initialY) {
+            this.initialY = initialY;
+            currentY = initialY;
+        }
+
+        Coordinates getCurrentCoordinates() {
+           return new Coordinates(currentX, currentY);
+        }
+
+        protected Cell getCurrentCell() {
+            return getCell(getCurrentCoordinates());
+        }
+
+        /**
+         * Monster tries to kill the player -- usually if he stepped on the cell with the player.
+         */
+        abstract protected boolean tryToKill();
+
+        /**
+         * Decreasing tickTo, but if it's 0, sets it to tickPer.
+         */
+        protected int decreaseTick(int tickTo, int tickPer) {
+            if (tickTo == 0) {
+                return tickPer - 1;
+            }
+
+            return tickTo - 1;
+        }
+    }
+
+    /**
+     * Class for storing cell's coordinates, where the cell on the bottom-left corner of the screen has
+     * (0, 0) coordinates.
+     */
+    protected static class Coordinates {
+        /**
+         * X coordinate.
+         */
+        private int x;
+
+        /**
+         * Y coordinate.
+         */
+        private int y;
+
+        public Coordinates(int x, int y) {
+            this.x = x;
+            this.y = y;
+        }
+
+        Coordinates(Coordinates other) {
+            this.x = other.x;
+            this.y = other.y;
+        }
+
+        int getX() {
+            return x;
+        }
+
+        int getY() {
+            return y;
+        }
+
+        void moveToVector(MazeGame.Command command) {
+            switch (command) {
+                case LEFT:
+                    x--;
+                    break;
+                case RIGHT:
+                    x++;
+                    break;
+                case DOWN:
+                    y--;
+                    break;
+                case UP:
+                    y++;
+                    break;
+                default:
+                    throw new IllegalArgumentException("wrong command to move");
+            }
+        }
+
+        boolean equals(Coordinates coordinates) {
```

Comment:
The standard "equals" method signature takes an Object as the argument. Would it make sense to do the same here? Then the class would be eligible to be used in collections. You need to provide hashCode as well though.

Reply:
Done with Lombok, @Data generates equals and hashCode with expected semantic.